### PR TITLE
[11.0][FIX] membership: infinite recursion in computed fields

### PR DESCRIPTION
**Pull request by sbidoul**
Same as #22568
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/membership/models/partner.py [hunk 1]
```diff
@@ -86,12 +86,9 @@ def _compute_membership_stop(self):
...
     def _compute_membership_cancel(self):
         for partner in self:
-            if partner.membership_state == 'canceled':
-                partner.membership_cancel = self.env['membership.membership_line'].search([
-                    ('partner', '=', partner.id)
-                ], limit=1, order='date_cancel').date_cancel
-            else:
-                partner.membership_cancel = False
+            partner.membership_cancel = self.env['membership.membership_line'].search([
+                ('partner', '=', partner.id)
```
Comment (mreficent) [inline addons/membership/models/partner.py]: While I am trying to understand all the code of the this file, I have a question. Why this line can't be
```python
('partner', '=', partner.associate_member.id or partner.id),('date_cancel','=',False)
```
?

Comment (fmdl) [reply addons/membership/models/partner.py]: maybe add a domain on the company (in case of multi company).

**Comment by tivisse** [comment]
Will be handled at https://github.com/odoo/odoo/pull/22568

Thanks.

**Comment by tivisse** [comment]
@robodoo r+

**Comment by pedrobaeza** [comment]
Hooray!

**Comment by sbidoul** [comment]
Thanks!

**Pull request closed without merging**